fgpyo/fasta/reference_set_builder.py

```diff
@@ -0,0 +1,174 @@
+"""
+Classes for generating Fasta files and records for testing
+----------------------------------------------------------------
+"""
+# import hashlib
+import os
+import textwrap
+from tempfile import NamedTemporaryFile
+from typing import List
+from typing import Optional
+
+# pylint: disable=W0511
+
+
+class ReferenceSetBuilder:
+    """
+    Builder for constructing one or more fasta records.
+    """
+
+    # The default asssembly
+    DEFAULT_ASSEMBLY: str = "testassembly"
+
+    # The default species
+    DEFAULT_SPECIES: str = "testspecies"
+
+    # Way to store instance of ReferenceBuilder
+    # TODO make something better than a list... probably
+    REF_BUILDERS: List["ReferenceBuilder"] = []
+
+    def __init__(
+        self,
+        assembly: Optional[str] = None,
+        species: Optional[str] = None,
+        line_length: int = 80,
+    ):
+        self.assembly: str = assembly if assembly is not None else self.DEFAULT_ASSEMBLY
+        self.species: str = species if species is not None else self.DEFAULT_SPECIES
+        self.line_length = line_length
+
+    def add(
+        self,
+        name: str,
+    ) -> "ReferenceBuilder":
+        """
+        Returns instance of ReferenceBuilder
+        """
+
+        builder = ReferenceBuilder(name=name, assembly=self.assembly, species=self.species)
+        self.REF_BUILDERS.append(builder)
+        return self.REF_BUILDERS[-1]
+
+    def writer_helper(
+        self,
+    ) -> None:
+        """ Place holder for helper funciton """
+        return None
+
+    def to_temp_file(
+        self,
+        delete_on_exit: Optional[bool] = True,
+        calculate_md5_sum: Optional[bool] = False,
```

Comment:
If you type these as `Optional[bool]`, that makes me think that you want three different behaviors: one when it's true, one when it's false, and one when it's `None`.  I don't understand when you want the latter in this function.  (reminded me of this: https://medium.com/code-thoughts/the-toilet-paper-dilemma-a1ba48411332)
```suggestion
        delete_on_exit:bool = True,
        calculate_md5_sum:bool = False,
```

Reply:
I fixed the incorrect use of `Optional[]` in commit cb8ac832a05ba4a3ad304edb2de19c57d6f9b8f3